services/worker-manager/src/providers/aws.js

```diff
@@ -81,21 +81,20 @@ class AwsProvider extends Provider {
     this._enqueue = cloud.enqueue.bind(cloud);
   }
 
-  async provision({workerPool}) {
+  async provision({workerPool, existingCapacity}) {
     const {workerPoolId} = workerPool;
 
-    if (!workerPool.providerData[this.providerId] || workerPool.providerData[this.providerId].running === undefined) {
+    if (!workerPool.providerData[this.providerId]) {
       await workerPool.modify(wt => {
         wt.providerData[this.providerId] = wt.providerData[this.providerId] || {};
```

Comment:
Speaking of eventual consistency, this is an unconditional modification and could end up deleting data some other process added at this time.  It might be better to repeat the conditional inside the `modify` callback:

```suggestion
        if (!workerPool.providerData[this.providerId]) {
          wt.providerData[this.providerId] = wt.providerData[this.providerId] || {};
        }
```

(I don't see this causing any issues, and it's unrelated to this PR, so no need to accept this change - it's just an easy way to show what I meant)